writing propagate instead of update + some other things (#18)
thesis20/thesis_paper: sections/csgcn/csgcn-chapter.tex

It is, however, important to note that the neighboring nodes for users now include side-information for users and not only items, which can be seen in each of the layers now receiving a normalized sum of both items and user side-information, rather than just the normalized sum of items.
The same goes for the neighbors of items.
\\

> There are a number of graph convolution layers where for each central node, information about neighbor nodes is being propagated.

Reviewer: ```suggestion
There are a number of graph convolution layers where, for each central node, information about neighbor nodes is being propagated.
```

> There are a number of graph convolution layers where where for each central node, information about neighbor nodes is being propagated.

Reviewer 2: where where wherewolf

$N_{un}$ denotes the set of neighbors for the neighbor $un$ of node $u$, and $N_{in}$ denotes the set of neighbors for the neighbor $in$ of node $i$.
This means that when neigbor nodes' representations are being propagated, the normalization is done with both the number of neighbors for the central node and the number of neighbors of the neighbor node that is currently having its reprentation propagated.
\\\\

> The convolution layer propagates information to user and item nodes' representations by summing the normalized representations of the neighbor nodes.
> All types of nodes are being propagated with information about their neighbors, but since only the user and item nodes' represenations are being used in the prediciton function for our model, their representations are the only ones that are returned from the convolution layer.
> The argument for not also returning side-information nodes and using these in the prediction function, is that through the convolution, the information from these nodes are integrated into the representation of user and item nodes.

Reviewer 2: ```suggestion
The convolution layer propagates information to user and item nodes' representations by summing the normalized representations of the neighbor nodes.
All types of nodes are being propagated with information about their neighbors, but since only the user and item nodes' representations are being used in the prediction function for our model, their representations are the only ones that are returned from the convolution layer.
The argument for not also returning side-information nodes and using these in the prediction function is that through the convolution, the information from these nodes is integrated into the representation of user and item nodes.
```

The different $e_{ic}$'s of a given item $i$ and context $C'$ are added to the item to signify its popularity in the given context.
After the addition of the $e_{ic}$'s to the item embedding $e_i$, the cosine similarity scores of the user embedding $e_u$ and the item-context additions are calculated and summed together to produce a score: 
$$\hat{y}_{uic} = \sum_{c \in C'} e_u^T(e_i+e_{ic})$$

> As shown, the context embeddings are not a part of the convolution layer, but are only used in the score prediciton, meaning that they are learned directly through their appearence in the score prediction function.
> This is further described in \Cref{subsec:csgcn_is_training}.

Reviewer 2: ```suggestion
As shown, the context embeddings are not a part of the convolution layer but are only used in the score prediction, meaning that they are learned directly through their appearance in the score prediction function.
This is further described in \Cref{subsec:csgcn_is_training}.
```
